metadata/snapshot.go

```diff
@@ -280,6 +282,42 @@ func (s *snapshotter) View(ctx context.Context, key, parent string, opts ...snap
 	return s.createSnapshot(ctx, key, parent, true, opts)
 }
 
+// checkIfSnapshotExists checks if a snapshot with the given `key` exists in the given namespace `ns`
+// If `checkAllNamespaces` is set to `true` it checks other namespaces
+// too. If the snapshot is found then returns `true` and the name of the namespace in which
+// it was found. Returns false otherwise.
+func (s *snapshotter) checkIfSnapshotExists(tx *bolt.Tx, ns, key string, checkAllNamespaces bool) (bool, string, error) {
+	bkt := getSnapshotterBucket(tx, ns, s.name)
+	if bkt == nil {
+		return false, "", fmt.Errorf("can not find snapshotter %q: %w", s.name, errdefs.ErrNotFound)
+	}
+
+	if tbkt := bkt.Bucket([]byte(key)); tbkt != nil {
+		return true, ns, nil
+	}
+
+	if checkAllNamespaces {
+		vbkt := tx.Bucket(bucketKeyVersion)
+		if vbkt == nil {
+			return false, "", fmt.Errorf("can not find version bucket")
+		}
+
+		// iterate through each namespace
+		vc := vbkt.Cursor()
+		for nk, _ := vc.First(); nk != nil; nk, _ = vc.Next() {
```

Comment:
It feels redundant to search the same namespace here as we did on 295-297 when we iterate through all of them